social_core/utils.py

```diff
@@ -112,7 +112,7 @@ def sanitize_redirect(hosts, redirect_to):
 def user_is_authenticated(user):
     if user and hasattr(user, 'is_authenticated'):
         if isinstance(user.is_authenticated, collections.Callable):
-            authenticated = user.is_authenticated()
+            authenticated = user.is_authenticated
         else:
             authenticated = user.is_authenticated
```

Comment:
Your change seems to make the entire if condition unnecessary b/c both branches become the same.

Reply:
as of Django 1.10 [release](https://docs.djangoproject.com/en/3.0/releases/1.10/#using-user-is-authenticated-and-user-is-anonymous-as-methods) `is_authenticated` as a method was deprecated and introduced as a property instead and was removed in Django 2.0 as mentioned in issue #439 

to get rid of the deprecation warning and remove its usage as a method, I think we can safely remove this `if else` condition.